hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableGenericRecordBuilder.java

```diff
@@ -29,25 +29,39 @@
 
     private final ClassDefinition classDefinition;
     private final Object[] objects;
-    private final boolean[] isOverWritten;
+    private final boolean[] isWritten;
     private final boolean isClone;
 
     public PortableGenericRecordBuilder(ClassDefinition classDefinition) {
         this.classDefinition = classDefinition;
         this.objects = new Object[classDefinition.getFieldCount()];
         this.isClone = false;
-        this.isOverWritten = null;
+        this.isWritten = new boolean[objects.length];
     }
 
     PortableGenericRecordBuilder(ClassDefinition classDefinition, Object[] objects) {
         this.classDefinition = classDefinition;
         this.objects = objects;
         this.isClone = true;
-        this.isOverWritten = new boolean[objects.length];
+        this.isWritten = new boolean[objects.length];
     }
 
+    /**
+     * @return newly created GenericRecord
+     * @throws HazelcastSerializationException if a field is not written when building with builder from
+     *                                         {@link GenericRecord.Builder#portable(ClassDefinition)} and
+     *                                         {@link GenericRecord#newBuilder()}
+     */
     @Override
     public GenericRecord build() {
+        if (!isClone) {
+            for (int i = 0; i < isWritten.length; i++) {
+                if (!isWritten[i]) {
```

Comment:
if i do cloneWithBuilder (all isWritten fields are false) and i only overwrite one field, some fields of isWritten will be left as false, right? and this check would fail? am i missing something here?

Reply:
If you do clone with the builder this check will not be done. see ` if (!isClone) {` at the beginning. 
I have a test that verifies the behavior.